src/hooks/analytics/useTrackScreen.ts

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { usePostHog } from 'posthog-react-native';
+
+/**
+ * Hook to track a screen view.
+ * @param screenName - Name of the screen to track.
+ * @param properties - Additional properties to track.
+ */
+const useTrackScreen = (screenName: string, properties?: any) => {
+  const postHog = usePostHog();
+
+  React.useEffect(() => {
+    if (!postHog) {
+      return;
+    }
+
+    postHog.screen(screenName, properties);
+  }, [postHog, properties, screenName]);
```

Comment:
Here the properties must be momoized from the caller otherwise if `properties` is a non primitive object this will trigger an event submition every time the screen rerenders because the object will be recreated and so the new `properties` object will be a new instance and this will cause the `useEfect` to be re-executed.
The `useEffect` re-execution is caused from the fact that the dependencies inside this list are checked with the `===` operator and on objects this will only check that they are the same instance aka the reference point to the same value in memory.